From a8843732263d255d8637977bb07d05e042e30a7f Mon Sep 17 00:00:00 2001
From: Shaveta Leekha <shaveta@freescale.com>
Date: Wed, 19 Sep 2012 17:46:03 +0000
Subject: [PATCH 069/227] b4860: Updated the device tree for b4860qds

enable SPI flash support
	update NOR info and NAND partitions
	delete SPI partitions

Signed-off-by: Shaveta Leekha <shaveta@freescale.com>
[Kevin: Original patch taken from fsl sdk 1.3.1
QorIQ-SDK-V1.3.1-SOURCE-20121220-yocto.iso.]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/powerpc/boot/dts/b4860qds.dts |   39 +++++++-----------------------------
 1 file changed, 7 insertions(+), 32 deletions(-)

diff --git a/arch/powerpc/boot/dts/b4860qds.dts b/arch/powerpc/boot/dts/b4860qds.dts
index 7497210..8e6f0f7 100644
--- a/arch/powerpc/boot/dts/b4860qds.dts
+++ b/arch/powerpc/boot/dts/b4860qds.dts
@@ -43,19 +43,17 @@
 
 
 	ifc: localbus@ffe124000 {
-		reg = <0xf 0xfe124000 0 0x1000>;
+		reg = <0xf 0xfe124000 0 0x2000>;
 		ranges = <0 0 0xf 0xe8000000 0x08000000
-			  2 0 0xf 0xffa00000 0x00010000>;
+			  2 0 0xf 0xff800000 0x00010000
+			  3 0 0xf 0xffdf0000 0x00008000>;
 
 		nor@0,0 {
 			#address-cells = <1>;
 			#size-cells = <1>;
 			compatible = "cfi-flash";
 			reg = <0x0 0x0 0x8000000>;
-			/* TODO: Update bank-width depending upon
-			 * on-board Flash
-			 */
-			bank-width = <1>;
+			bank-width = <2>;
 			device-width = <1>;
 		};
 
@@ -85,9 +83,9 @@
 				label = "NAND Linux Kernel Image";
 			};
 
-			partition@e00000 {
+			partition@c00000 {
 				/* 500MB for Root file System Image */
-				reg = <0x00e00000 0x1F400000>;
+				reg = <0x00c00000 0x1F400000>;
 				label = "NAND RFS Image";
 			};
 		};
@@ -106,35 +104,12 @@
 		ranges = <0x00000000 0xf 0xfe000000 0x1000000>;
 		reg = <0xf 0xfe000000 0 0x00001000>;
 		spi@110000 {
-			status = "disabled";
 			flash@0 {
 				#address-cells = <1>;
 				#size-cells = <1>;
-				compatible = "spansion,s25sl12801";
+				compatible = "sst,sst25wf040";
 				reg = <0>;
 				spi-max-frequency = <40000000>; /* input clock */
-				partition@0 {
-					/* This location must not be altered  */
-					/* 1MB for u-boot Bootloader Image */
-					label = "SPI U-Boot Image";
-					reg = <0x00000000 0x00100000>;
-					read-only;
-				};
-				partition@100000 {
-					/* 5MB for Linux Kernel Image */
-					label = "SPI Linux Kernel Image";
-					reg = <0x00100000 0x00500000>;
-				};
-				partition@600000 {
-					/* 1MB for DTB Image */
-					label = "SPI DTB Image";
-					reg = <0x00600000 0x00100000>;
-				};
-				partition@700000 {
-					/* 9MB for RFS Image */
-					label = "SPI RFS Image";
-					reg = <0x00700000 0x00900000>;
-				};
 			};
 		};
 
-- 
1.7.9.7

